### PR TITLE
[JN-378] Add basic survey editor

### PR DESCRIPTION
**Pull request by nawatts**
This replaces the SurveyJS editor in the admin UI with our own basic editor. Currently, it allows editing the survey as JSON and viewing a preview of the survey rendered with SurveyJS.

### CODE REVIEW

File: ui-admin/src/App.css [hunk 1]
```diff
@@ -86,4 +85,10 @@ a {
   max-height: 90vh;
 }
 
-
+/* Have survey editor tabs fill page height */
```
Comment (devonbush) [inline ui-admin/src/App.css:88]: I'm surprised the bootstrap library doesn't have a way to specify these as inline styles

File: ui-admin/src/study/surveys/editor/SurveyPreview.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+export const SurveyPreview = (props: SurveyPreviewProps) => {
+  const { survey } = props
+
+  const [surveyModel] = useState(() => {
+    const fakeForm = { content: JSON.stringify(survey) } as Survey
+
+    const model = new SurveyModel(extractSurveyContent(fakeForm))
+    model.focusFirstQuestionAutomatic = false
+    model.showTitle = false
```
Comment (devonbush) [inline ui-admin/src/study/surveys/editor/SurveyPreview.tsx:19]: showTitle = false and widthMode static are also set on the participant side -- do we ever not want these?  And if we don't, should we pull these into a "configureDefaultSurveyDisplay" utility method somewhere?

**Review by devonbush — APPROVED** [review]
Like the approach and structure of getting just the basic json editor in first.  Couple of questions, nothing blocking or major

File: ui-admin/src/App.css (hunk 1 above)
Comment (nawatts) [reply ui-admin/src/App.css:88]: It does, but react-bootstrap doesn't expose a way to add classes to the `.tab-content` element.

File: ui-admin/src/study/surveys/editor/SurveyPreview.tsx (hunk 2 above)
Comment (nawatts) [reply ui-admin/src/study/surveys/editor/SurveyPreview.tsx:19]: I'm thinking `extractSurveyContent` may become a FormContent => SurveyModel function that applies this sort of shared configuration to the SurveyModel.

File: ui-admin/src/study/surveys/SurveyEditorView.tsx [hunk 3]
```diff
@@ -1,63 +1,77 @@
...
+              'btn btn-primary me-md-2',
+              { disabled: !isSaveEnabled }
+            )}
+            type="button"
+            onClick={onClickSave}
+          >
             Save
           </button>
         )}
         <button className="btn btn-secondary" type="button" onClick={onCancel}>Cancel</button>
```
Comment (breilly2) [inline ui-admin/src/study/surveys/SurveyEditorView.tsx:63]: I noticed that cancel is navigating back one level further than I think it should. Looks like the `onCancel` function being passed in from `SurveyView` is responsible for this. But it looks like this isn't new with this PR.

**Review by breilly2 — APPROVED** [review]

**Pull request merged**